File: app/src/main/java/com/zhiyong/tingxie/ui/question/QuestionViewModel.java
package com.zhiyong.tingxie.ui.question;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.annotation.NonNull;

import com.zhiyong.tingxie.db.Term;
import com.zhiyong.tingxie.ui.main.QuizRepository;

import java.util.List;

public class QuestionViewModel extends AndroidViewModel {
    private QuizRepository mRepository;
    private LiveData<List<Term>>  mQuestionItems;

    public QuestionViewModel(@NonNull Application application, long quizId) {
        super(application);
        mRepository = new QuizRepository(application, quizId);
        mQuestionItems = mRepository.getRemainingQuestionsOfQuiz();
    }

    LiveData<List<Term>> getRemainingQuestions() {
        return mQuestionItems;
    }
}

File: app/src/main/java/com/zhiyong/tingxie/ui/question/QuestionViewModel.kt
package com.zhiyong.tingxie.ui.question

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import com.zhiyong.tingxie.QuizRepository
import com.zhiyong.tingxie.network.NetworkWordItem
import com.zhiyong.tingxie.viewmodel.CrudStatus
import kotlinx.coroutines.launch

class QuestionViewModel(application: Application, quizId: Long) :
  AndroidViewModel(application) {
  private val mRepository: QuizRepository = QuizRepository(application)

  private val _notCorrectWordsRandomOrder = MutableLiveData<List<NetworkWordItem>>()
  val notCorrectWordsRandomOrder: LiveData<List<NetworkWordItem>> =
    _notCorrectWordsRandomOrder

  private val _notCorrectWordsRandomOrderStatus = MutableLiveData<CrudStatus>()
  val notCorrectWordsRandomOrderStatus: LiveData<CrudStatus> =
    _notCorrectWordsRandomOrderStatus

  init {
    getNotCorrectWordsRandomOrder(quizId)
  }

  private fun getNotCorrectWordsRandomOrder(quizId: Long) {
    viewModelScope.launch {
      _notCorrectWordsRandomOrderStatus.value = CrudStatus.LOADING
      try {
        _notCorrectWordsRandomOrder.value =
          mRepository.notCorrectWordsRandomOrder(quizId)
        _notCorrectWordsRandomOrderStatus.value = CrudStatus.DONE
      } catch (e: Exception) {
        _notCorrectWordsRandomOrderStatus.value = CrudStatus.ERROR
      }
    }
  }
}
